engine/src/test/java/com/redhat/cloud/notifications/InventoryTestHelpers.java

```diff
@@ -82,6 +84,79 @@ public static EmailAggregation createEmailAggregation(String tenant, String bund
         return aggregation;
     }
 
+    /**
+     * Creates a minimal email aggregation, and fills any other fields either
+     * by providing the given {@code inventoryId} or by prefixing that ID with
+     * the name of the field.
+     * @param eventType the type of the event to simulate the aggregation from.
+     * @param inventoryId the inventory ID of the event.
+     * @param displayName the display name of the host.
+     * @return the generated email aggregation.
+     */
+    public static EmailAggregation createMinimalEmailAggregationV2(
+        String eventType,
+        UUID inventoryId,
+        String displayName
+    ) {
+        final EmailAggregation aggregation = new EmailAggregation();
+        aggregation.setBundleName("rhel");
+        aggregation.setApplicationName("inventory");
+        aggregation.setOrgId(DEFAULT_ORG_ID);
+
+        final Action emailActionMessage = new Action();
+        emailActionMessage.setBundle("rhel");
+        emailActionMessage.setApplication("inventory");
+        emailActionMessage.setTimestamp(LocalDateTime.now());
+        emailActionMessage.setEventType(eventType);
+        emailActionMessage.setOrgId(DEFAULT_ORG_ID);
+
+        // Transform the tags to the expected format.
+        record Tag(String value, String key) { }
+
+        final Map<String, String> tags = Map.of("inventory-id", inventoryId.toString());
+
+        final List<Tag> convertedTags = new ArrayList<>();
+        for (Map.Entry<String, String> entry : tags.entrySet()) {
+            convertedTags.add(
+                new Tag(entry.getKey(), entry.getValue())
+            );
+        }
+
+        emailActionMessage.setContext(
+            new Context.ContextBuilder()
+                .withAdditionalProperty("inventory_id", inventoryId)
+                .withAdditionalProperty("hostname", String.format("hostname-%s", inventoryId))
```

Comment:
Could we keep only usefull properties from `context` and `payload` to keep it simple?

Reply:
I was leaving this as a form of "here's the payload that we're expecting from Inventory", in case we need to add stuff in the future, but yes, I guess we can always look back at the docs if we need it.

Reply:
Thanks, because tenants are free to change payload content at any time (then validate them on stage), even if your sample is accurate today, it could not be in a month. this is why I prefer keeping only used data.

Reply:
Understood! Thank you :smile: :+1: